core/src/main/java/org/hisp/dhis/android/core/trackedentity/internal/TrackerQueryFactory.kt

```diff
@@ -67,8 +67,11 @@ internal abstract class TrackerQueryFactory<T, S : TrackerBaseSync>(
                         internalFactory.queryGlobal(globalPrograms)
                 }
             }
-        } else {
+        } else if (params.program() != null) {
             internalFactory.queryPerProgram(params.program())
+        } else {
+            val programs = params.programStageWorkingLists()?.map { it.program().uid() }
```

Comment:
Same here, it should consider the program uids coming from `trackedEntityInstanceFilters`, `eventFilters` and `programStageWorkingLists`